### ISSUE TITLE
Port twisted.application.internet on py3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#7817]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)| @adiroiban reported|
|-|-|
|Trac ID|trac#7817|
|Type|enhancement|
|Created|2015-03-16 11:10:39Z|
|Branch|https://github.com/twisted/twisted/tree/twapp-internet-py3-7817|

<details><summary>Searchable metadata</summary>

```
trac-id__7817 7817
type__enhancement enhancement
reporter__adiroiban adiroiban
priority__normal normal
milestone__Python_3_x Python-3.x
branch__branches_twapp_internet_py3_7817 branches/twapp-internet-py3-7817
branch_author__hawkowl hawkowl
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1426504239244166 1426504239244166
changetime__1435820583269337 1435820583269337
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_1"></a>|@hawkowl commented|
|-|-|

(In [45158]) Branching to twapp-internet-py3-7817.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_2"></a>|@hawkowl commented|
|-|-|

This is an initial port; the missing coverage is in ```twisted.test.test_application```, which needs ```twisted.application.app``` to be ported, which will be put up for review right after this one is merged.

Builders are spun and looking good, please review.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_3"></a>|@adiroiban set owner to @hawkowl|
|-|-|

Changes look good.

Why have you removed test_applicationComponentsArePersistable ?

Please check my commend and merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_4"></a>|@hawkowl set status to `closed`|
|-|-|

(In [45163]) Merge twapp-internet-py3-7817: Port twisted.application.internet to Python 3

Author: hawkowl
Reviewer: adiroiban
Fixes: [#7817](https://github.com/twisted/twisted/issues/7817)
